tests/test_transforms.py

```diff
@@ -7,56 +7,88 @@
+class TestVolumetricTransform:
+    """Unit tests for volumetric transformations using `_vol_to_vol`."""
 
-        This method initializes the file paths used in all subsequent tests.
+    # Source images
+    t1w_source = Path(
+        "/Users/tamsin.rogers/Desktop/github/neuromaps/"
+        "share_with_T1w/Inputs/D99/src-D99_res-0p25mm_T1w.nii"
+    )
+    label_source = Path(
+        "/Users/tamsin.rogers/Desktop/github/neuromaps/"
+        "share_with_T1w/atlas/D99_atlas_v2.0.nii"
+    )
 
-        Attributes:
-            source_file: Path to the source volumetric file to be transformed.
-            target_same: Path to the target volumetric file with the same resolution.
-            target_diff: Path to the target volumetric file with a different resolution.
-        """
-        self.source_file = Path(
-            "/Users/tamsin.rogers/Desktop/github/neuromaps/share_with_T1w/Inputs/D99/src-D99_res-0p25mm.T1w.nii"
-        )
-        self.target_same = Path(
-            "/Users/tamsin.rogers/Desktop/github/neuromaps/share_with_T1w/Inputs/NMT2Sym/src-NMT2Sym_res-0p25mm.T1w.nii"
-        )
-        self.target_diff = Path(
-            "/Users/tamsin.rogers/Desktop/github/neuromaps/share_with_T1w/Inputs/MEBRAINS/src-MEBRAINS_res-0p40mm.T1w.nii"
-        )
+    # Target images
+    target_same = Path(
+        "/Users/tamsin.rogers/Desktop/github/neuromaps/"
+        "share_with_T1w/Inputs/NMT2Sym/src-NMT2Sym_res-0p25mm_T1w.nii"
+    )
+    target_diff = Path(
+        "/Users/tamsin.rogers/Desktop/github/neuromaps/"
+        "share_with_T1w/Inputs/MEBRAINS/src-MEBRAINS_res-0p40mm_T1w.nii"
+    )
 
     @pytest.mark.parametrize("target_attr", ["target_same", "target_diff"])
-    def test_vol_to_vol_resolution(self, target_attr: str) -> None:
-        """Test that `_vol_to_vol` produces a file matching the target resolution.
+    @pytest.mark.parametrize("interp", CONTINUOUS_INTERPS)
+    def test_vol_to_vol_continuous(self, target_attr: str, interp: str) -> None:
+        """Test continuous interpolators for volumetric transforms."""
+        target_file = getattr(self, target_attr)
 
-        Args:
-            target_attr: Attribute name of the target file to test against.
-                Can be either `'target_same'` or `'target_diff'`.
+        # Interpolators that should raise NotImplementedError
+        if interp in {"gaussian", "BSpline"}:
+            with pytest.raises(NotImplementedError):
+                _vol_to_vol(self.t1w_source, target_file, interp=interp)
+            return
 
-        Raises:
-            AssertionError: If the output file does not exist, is not a NIfTI image,
-                or if its resolution does not match the target file.
-        """
-        target_file = getattr(self, target_attr)
-        result_path = _vol_to_vol(self.source_file, target_file)
+        # All other continuous interpolators should run normally
+        result = _vol_to_vol(self.t1w_source, target_file, interp=interp)
```

Comment:
It would be better here and below to split up the tests that you expect to raise an error vs those that don't.